crates/sargon/src/profile/mfa/security_structures/security_shield_builder.rs

```diff
@@ -459,6 +459,31 @@ impl SecurityShieldBuilder {
     }
 }
 
+// ================
+// ==== STATIC ====
+// ================
+impl SecurityShieldBuilder {
+    pub fn sorted_factor_sources_for_selection(
+        factor_sources: Vec<FactorSource>,
+    ) -> Vec<FactorSource> {
+        let mut sorted = factor_sources;
+        sorted.sort_by_key(|fs| fs.factor_source_kind().display_order());
```

Comment:
the perpetual naming challenge for these kind of things... :) I think `let mut sorted = factor_sources` is misleading, since it is in fact not sorted yet!

Luckily In Rust we have ownership!

so:
```rust
let mut unsorted = factor_sources;
unsorted.sort....
let sorted = unsorted;
// no `unsorted` can never be referenced again! it has been consumed! Rust owns!
sorted
```